### PR TITLE
Cytology Update + Redone Metastation Science

### PR DESCRIPTION
**Pull request by AlinaStarkova**
THE MAP FUNCTIONS AND EVERY CHANGE HAS BEEN TESTED AND ALL WORK

## About The Pull Request

Moderate Update to make Cytology viable, Note: Everything changed and added has been tested, everything functions as intended and nothing has been broken (note that swabs cant be directly clicked into the biobag, you have to open the biobag then click the swab into one of the slots directly)

1. Science has been completely redone on Metastation. (rework by Zytolg, setup of disposals and some other specific edits to get science connected again done by me)
image

image

2.a. Added a fully connected and functional Cytology Lab to Xenobio.
image

3. Added an independent Cytology Area
image

4. Allowed Biobags to hold Petridishes and Swabs
40620de418fc1af7343379943af10cb6

5. Allowed Labcoats to hold Biopsy Tools
6. Created a new Smart Fridge Variant: Petri Dish Fridge, which can only store Petridishes, Metastation Cytology Lab starts with 5 Dishes in its fridge (currently it stacks all petridishes together no matter what, as petridishes dont get unique names, thats an issue with cytology itself, i was in the middle of coding a fix for this by having random ID tags assigned to the dishes, however dreamdaemon took a crap on me and wont resolve itself, so thats on hold, or perhaps someone else can fix this? i was thinking when a new dish is made, its given a random # between 0-1000 after its name, nice and simple)
5055c21833aab7c144e45f5a4ac252eb

7. Added and Sprited Cytology Wikibook
image

## Why It's Good For The Game

Cytology was added to the game as well as plumping for said Cytology, however there currently is not enough space to fully utilize Cytology at the current point in time (especially since Xenobiologists tend to end up creating many extra pens). This update not only adds a dedicated Cytology work lab (only on metastation at the moment) but also updates functionality to act as a quality of life benefit for those wishing to work in Cytology

## Changelog
:cl: Nari Harimoto
add: Petri Dish Smart Fridge
add: Cytology Area
add: Cytology Lab (Metastation)
tweak: Completly redone Science Department on Metastation (By Zytlog)
tweak: Labcoats can now carry Biopsy Tools
tweak: Biobags can now carry Swabs and Petridishes
add: Cytology Wikibook
/:cl:

### CODE REVIEW

**Comment by Rustybuckets6601** [comment]
Looks great, nice job

**Comment by AlinaStarkova** [comment]
> 
> 
> Looks great, nice job

my area sprite looks way off, but uh, i aint a spriter so sue me xD

**Comment by Fikou** [comment]
the code tweak stuff is cool
that sprite looks dumb we literally have an extremely easy font system for consistency that every single other area uses and you decide to not use it
the map kinda sucks its a giant rectangle that doesnt fit in nicely with anything, and seems really annoying to have another set of bolted doors in xenobio

**Comment by RaveRadbury** [comment]
Cytology was added for xenobio content, why are you splintering it into yet another unlisted science job?

**Comment by AlinaStarkova** [comment]
> that sprite looks dumb we literally have an extremely easy font system for consistency that every single other area uses and you decide to not use it
i didnt "decide" not to use it, im not a spriter, it was actually quite difficult for me to make that, regardless, its a backend sprite and you wont ever see it in game

as for the map, xenobio as a whole is a big box, literally, atleast with this its not a complete box (i did add a small maint area for cytology)
fdc48b3d5dffe856c7f8a1ca3f5ca470








> Cytology was added for xenobio content, why are you splintering it into yet another unlisted science job?
its literally not an "unlisted job" its a new room for it, just like Chemistry got the Pharmacy Room, Xenobio lab does not have enough space for both xenobio and cytology, (Cytology has plumping, dontcha know?) just like chemistry was given a lab to work their plumping, xenobiology does as well, the "Cytology Area" is just for mapping purposes (AI or suit sensors will show "Cytology Lab" just as the chemistry lab shows "Pharmacy")

**Comment by Fikou** [comment]
> i didnt "decide" not to use it, im not a spriter, it was actually quite difficult for me to make that, regardless, its a backend sprite and you wont ever see it in game

no effort is needed to go to the font image in the dmi and copy the letters and arrange them together, you are defeating the whole purpose of the area icons and how they are made

adding some maint doesnt make it not a big ugly tumor on the xenobio lab with a secondary bolting system

**Comment by AlinaStarkova** [comment]
> no effort is needed to go to the font image in the dmi and copy the letters and arrange them together, you are defeating the whole purpose of the area icons and how they are made

i didnt even know you can do that that, i simply opened it in dream maker and the tools i was given was pretty limited, i will try to fix the sprite

> adding some maint doesnt make it not a big ugly tumor on the xenobio lab with a secondary bolting system

again, metastation xenobio is already a big square, and its floating off into space with only 1 hall connection, there was little i could effectively do to make the cytology lab more organic, I can however try to "connect" the whole xeno/cytology labs to the station by creating a maint area around it. im very open to suggestions, so instead of just complaining, give me ideas

File: strings/locations.json [hunk 1]
```diff
@@ -92,6 +92,7 @@
         "Virology",
         "Warehouse",
         "Waste Disposal",
-        "Xenobiology Lab"
+		"Xenobiology Lab",
+		"Cytology Lab"
```
Comment (Jared-Fogle) [inline strings/locations.json:96]: fix the indentation

**Comment by Jared-Fogle** [comment]
why would you need a second set of cycling airlocks for cytology

**Comment by AlinaStarkova** [comment]
fixed the area sprite font to match the standard

43f80401789521a43008505a5b82ee98

**Comment by AlinaStarkova** [comment]
> 
> 
> why would you need a second set of cycling airlocks for cytology

i felt like it was needed for the same reason xenobio and virology need them inherently, cell work and slime work are inherently two different things, and im not well versed in Cytology, so i mapped on the assumption there could be biohazard risks involved (or could be in the future) so an extra layer of protection, it doesnt inherently detract from space anyway because i was told cytology doesnt need as much space as pharmacy. im still trying to figure out how to resolve the blockiness inherent with metastation xenobio as a whole, im not entirely sure what im supposed to do about that since xenobio is completely disconnected from the rest of the station, i could maybe try merging xenobio with the station, or maybe move the solars farther east and push cytology on top of xenobio perhaps? im open to suggestions

**Comment by AlinaStarkova** [comment]
Maybe this variant is better?? i moved xenobio down a bit further and added cytology up top, so instead of sticking off the side its completely a part of xenobio sandwiched between it and the station @Jared-Fogle @Fikou 

image

**Comment by D4C-420** [comment]
Cytology can be done with as little as 2 tiles allocated for plumbing. One input gate for manfully inputting regents and one vat for growing them. The 3 largest problems I find playing cytology is one, it can nearly half an hour to grow anything. 2 the pay off for growing that thing is often times lack luster (once say some poor sod spend that half an hour trying to grow a roach. For reference a fast and capable xenobio can get golds in 20-30 minutes.). The only unique set of mobs cytology can grow are those clown ones. Outside of learning about it from other people or coming here to git hub there is zero way to learn about the only real reason to do cytology. This brings me to my 3rd problem with cytology and that's information on how to actually do it is not shown and communicated in any way. The wiki page as is tells you nothing beyond vague hints of how to do it and there is not way to learn how to do cytology in game (not even a book like most other complex jobs/tasks get). 

tldr. The things making cytology unviable is not an issue with space but the time investment to do it each round and the end result is not worth it and barely anyone knows how to do it because we tell players nothing on how to do it both in game and out of game. 

possible solutions: to 1 : make the growing time significantly less and make the regents for "common" (non cytology exclusive eg roaches, slimes, things found on station round start so not lava land mobs if they can even be swabbed)less of a pain in the ass to get. This cuts down on time taken to grow mobs that really are not worth the effort. 2. Make it more rewarding. Taking so long to make a couple roaches is not a good incentive to play cytology. Making a couple clown abominations is. The simplest and most difficult way to make cytology viable is to add more unique mobs to it. A more creative way is to let you modify the pre-existing creations by using chemicals found in simple plumbing setups (eg giving a roach more health or speed for growing it with say oil or some other easy to make chem or using it to change their colors.). The solution to 3 is update the wiki page at the very least and add a book for it in the cytology area"(like viro gets for example).

**Comment by AlinaStarkova** [comment]
> 
> 
> Cytology can be done with as little as 2 tiles allocated for plumbing. One input gate for manfully inputting regents and one vat for growing them. The 3 largest problems I find playing cytology is one, it can nearly half an hour to grow anything. 2 the pay off for growing that thing is often times lack luster (once say some poor sod spend that half an hour trying to grow a roach. For reference a fast and capable xenobio can get golds in 20-30 minutes.). The only unique set of mobs cytology can grow are those clown ones. Outside of learning about it from other people or coming here to git hub there is zero way to learn about the only real reason to do cytology. This brings me to my 3rd problem with cytology and that's information on how to actually do it is not shown and communicated in any way. The wiki page as is tells you nothing beyond vague hints of how to do it and there is not way to learn how to do cytology in game (not even a book like most other complex jobs/tasks get).
> 
> tldr. The things making cytology unviable is not an issue with space but the time investment to do it each round and the end result is not worth it and barely anyone knows how to do it because we tell players nothing on how to do it both in game and out of game.
> 
> possible solutions: to 1 : make the growing time significantly less and make the regents for "common" (non cytology exclusive eg roaches, slimes, things found on station round start so not lava land mobs if they can even be swabbed)less of a pain in the ass to get. This cuts down on time taken to grow mobs that really are not worth the effort. 2. Make it more rewarding. Taking so long to make a couple roaches is not a good incentive to play cytology. Making a couple clown abominations is. The simplest and most difficult way to make cytology viable is to add more unique mobs to it. A more creative way is to let you modify the pre-existing creations by using chemicals found in simple plumbing setups (eg giving a roach more health or speed for growing it with say oil or some other easy to make chem or using it to change their colors.). The solution to 3 is update the wiki page at the very least and add a book for it in the cytology area"(like viro gets for example).

i didnt make cytology so im not going to try to mess with its overall functions, im simply adding quality of life functions to make it easier to do, and yes, space does seem to be one, considering ive had 3+ people complain about space for the plumping and i was told this is a good size (perhaps cytology would work faster if you have multiple vats running at a time? hmmm)

**Comment by AlinaStarkova** [comment]
made it more organic, hopefully this design appeases everyone, or most people. i have to sleep

image


image

**Comment by Krysonism** [comment]
On topic: I like the idea of this PR, i just wish it was a bit more integrated with xenobio, perhaps by replacing some of the walls with windows?

Responding to comments:

>The only unique set of mobs cytology can grow are those clown ones. Outside of learning about it from other people or coming here to git hub there is zero way to learn about the only real reason to do cytology.

There are actual 3 very powerful mobs unique to cytology outside of the clown ones: sholean grapes, vat beast and gelatinous cube.

I have more mobs planned and partially sprited.

>A more creative way is to let you modify the pre-existing creations[...]

This is also planned in the form of chimaeric splices.

**Comment by AlinaStarkova** [comment]
> 
> 
> On topic: I like the idea of this PR, i just wish it was a bit more integrated with xenobio, perhaps by replacing some of the walls with windows?

i already did that and updated the images 10 minutes ago, as per suggestions in the discord, i even now just added blast doors and firelocks on the inner windows and doors since *apparently* 90% of the mobs you make there try to kill you? atleast thats what i was told

image

**Comment by AlinaStarkova** [comment]
Added and sprited Cytology wikibook

image

**Comment by D4C-420** [comment]
yee, nice work

**Comment by AlinaStarkova** [comment]
I was going to update the petri dish code so whenever you create a new sample it renames the dish with a number after it so that its seperated in the fridge, however dream daemon no longer works for me for some reason, i think an antivirus or my ISP or something quarantined it however theres no logs saying they did. so i cant actually test any coding i do. sad, maybe someone else can worry about that feature? i hope it works again relatively soon

**Comment by Cartographer-D** [comment]
Important Fixes:
image
Having a "maintenance" area that is not actually maintenance (Does not protect from radiation) is inconsistent and bad practice. You can also accomplish the same effect (Having an area that can't be seen well) in a better way.
image
Having both a door and window lockdown is also incredibly unnecessary. I can understand the privacy shutters, but there is no reason to restrict access to the door. Remember, we plan for departments to have flaws.
image
Having an extra toolbox in this space is not necessary and power creep.
image
This is a lot of extra, unnecessary space that can be moved down easily.
image
Corner machines surrounded by tables are generally a no go, as you can't easily move them or change them.
image
Cameras on windows aren't consistent and look bad.
image
Same with cameras over machines (The machines are designed to look like they take up the whole space.)
image
This camera is also improperly tagged (Actually, they all are)
image
You don't need two intercomms, one will work fine.
image
You want to minimize pathing for atmos
image
You don't want to put pipes through walls if you can help it. It would be better to reroute through the door.
image
Inner departmental walls do not need to be reinforced.

Overall, the space is still too large as well.
And your space walkway bumps right against the wall of the station, very bad.

Good stuff:
It looks like you got in all the objects that were needed. This is a BIG plus, especially since most of the varedits are done correctly.
The design looks organic and has pretty good sight.
Making its own area is great so long as you fix the font and such.


Here's a new layout that fits all the criteria you need:
image

If you want, I can PR to your branch and help you fix these problems.

**Comment by AlinaStarkova** [comment]
ye, i havent gotten around to adding cytology maint, it feels unnesscary, ill just add it as xenobio maint

the toolbox is because i was told cytology needs a wrench and i didnt see the harm in a full toolbox, science already has like 3 so its not like its an advantage to add one more but closer to the workplace, but ill replace that with just a wrench

door privacy shutter was because the door is actually a window door, so mobs can see through it and continue aggression the same way they can on the window, and unlike the walls the mobs can break doors just like windows, yet i didnt want them on the same button so you can close the windows but still get in and out seperately, if thats a particularly big issue i can combine their buttons

as for the extra space, i already have plans to add things there

as for the corner machine, fridges are rarely messed with or moved, and the table is glass so easily removed if required, i dont think thats much of a concern (icebox also has their xenobio fridges in a corner surrounded by a table and a console)

the cameras i have noticed their placement was bad and plan to move them, thanks for the tag tho, i forgot about that

as for 2 intercoms, i noticed bigger rooms have multiple (xenobio itself has like 3) because they have a certain range, granted the left side doesnt have anything that would make you work there for long

atmos pathing was a remnant from another position of the room, i copy pasted it over and edited and shrunk it, ill fix atmos as well

same with chute pipe, remnant from another position, ill reroute that too

too large is better than too little, plus ive been told cytology has plans to be expanded, so im preparing for that

as for the walkway, thats legitimately not me, i didnt move the walkway, i simply expanded it down on the lower half when i moved xenobio down.... so thats whoever mapped that long before me, I will fix that though

thank you for the suggestions, everyone else just complains on my PR without actually saying what they would prefer or what should change lol

**Comment by Cartographer-D** [comment]
"as for the extra space, i already have plans to add things there"
Don't say the map is done and tested if you have plans to add things still.

"too large is better than too little, plus ive been told cytology has plans to be expanded, so im preparing for that"
In the last photo, all of that stuff there on the left is fluff space, meaning it can be redone into something else. It can even be expanded to 4x4 instead of 3x4. You can also add stuff to the bottom near the windows.

"science already has like 3 so its not like its an advantage to add one more"
Exactly my point, power creep.

"ye, i havent gotten around to adding cytology maint, it feels unnesscary, ill just add it as xenobio maint"
Don't add maint at all, especially if it means making the same area be in two different places. There is already a section of maint that they have access to. The pen should be enough of a secluded spot for the other stuff.

"fridges are rarely messed with or moved"
The point is to be consistent. Icebox is just bad, don't use it as a reference. I might go in and fix that on there myself but I hate box.

**Comment by Cartographer-D** [comment]
Also, there seems to be a space missing in the area just above xenobio in the abandoned garden. If you add an area space to there so it isn't spaced, that would be fine.

**Comment by AlinaStarkova** [comment]
reee the map was done and tested, but people didnt like what i had and kept telling me little things to change, so ive had to change things, and still have things to change. ill get to it. 😢

ok, ill change that little maint to closet instead. im adding a secure pen to cytology, so space will be tightened up.

**Comment by Cartographer-D** [comment]
I edited my comment to include stuff about the door shutter, also.

**Comment by AlinaStarkova** [comment]
Updated the design for Cytology Lab to use up the extra space and have some more function
Expanded the xenobio-sci maint airlock
Rearranged the xenobio walkway 
Updated the cytology lab area font
image



image
image
image

File: _maps/map_files/KiloStation/KiloStation.dmm [hunk 2]
```diff
@@ -12704,7 +12704,7 @@
 	height = 22;
 	id = "whiteship_home";
 	name = "SS13: Auxiliary Dock, Station-Port";
-	roundstart_template = /datum/map_template/shuttle/mining_common/kilo;
+	roundstart_template = null;
```
Comment (Fikou) [inline _maps/map_files/KiloStation/KiloStation.dmm:12707]: unrelated change?

Comment (AlinaStarkova) [reply _maps/map_files/KiloStation/KiloStation.dmm:12707]: person who made the map told me it was an incorrect var and it was causing issues some people have noticed (logically, theres a shuttle docked where the white ships supposed to dock). i said i would just add that in since im doing a map edit anyway, so not *completely* unrelated. i can open up a whole PR just for that one line though if need be

Comment (Fikou) [reply _maps/map_files/KiloStation/KiloStation.dmm:12707]: also it already takes null from the parent doesnt it

Comment (AlinaStarkova) [reply _maps/map_files/KiloStation/KiloStation.dmm]: i have zero clue, but null is what that var is by default, and is also what it was given when i hit the var reset button.

i moved it to its own 1 line PR

**Comment by Cartographer-D** [comment]
Now that Zytolg's changes have been added on to this PR, I have more feedback:

First, continually adding things on to a PR, especially without proper documentation/comments and vastly changing scope, is generally considered bad form. People who originally viewed the PR might not be aware of the changes you made. People also might like some of the things and not all of the things, and it's basically like you're holding hostage some good stuff to make sure you get everything in. For example, if people didn't like Zytolg's redo of science or he didn't want to fix changes, they would have to choose between having the cool cytology stuff or keeping everything the same. There's a number of other problems you can run into doing this, so please try not to do this in the future.

Now for the map feedback:

Overall, the layout is fairly well done minus the very long hallways. I think you did a decent job of breaking it up, I just wish it was shorter. That can't be changed without completely digging it up, though, so we'll ignore it. I think you should swap genetics and the recharge bay. There are also a few glaring issues I get into later.


**Maintenance**
image
Having a roundstart grille on wire in maintenance is very poor form. Please remove it.

image
This is a good throwback. Try to add some more stuff in here though. Right now it's very bland. Maybe you could add spare science supplies to it?

image
This piping is connected to a wall and nothing else. If it's intentional that it's not connected, make it more obvious by having it run up a tile or two so people can see it.

image
You should always be cautious about having these very long maintenance areas with no way out unless you have access. Consider moving the area around or replacing the extra firing range with maintenance since you already have a room there for testing things (the reinforced floor area).

**Major Areas**
image
This is a very good design for the RD Lab. Not too hard to get in if you have departmental access, but still very secure. Having a sec post that can actually see what's happening in the department is great, too. I notice the shutters on the north windows; consider adding a button on security's side to open the shutters back up as a counter to forting.

image
I'm worried about the public desk being too far away from the working area. This workplace notoriously has problems trying to get the attention of people inside it. Consider moving the lath and printers up, especially since you can shutter the windows if you're doing something suspicious. It has shutters on doors, which I explain below in better detail.

The roboticist's is great and I approve everything about it. Great work. Especially like the touch with the tinted windows. Might consider replacing the topmost tinted windows in the recharge bay.

image
This geneticist's office looks great. There are three problems with it, though. First, it's more fortified than even the research director's office, you should replace some of the rwalls with normal walls and ABSOLUTELY remove the shutter from over the door (Think about if the armory had a shutter over it's door). Second, it's pretty far down the departures hallway. If you really want it to interact with the crew, swap its place with the recharge bay. Third, the decaling is completely different from the rest of the department. I think you should consider doing a decal overhaul for the rest of the department rather than genetics, because this looks really good.

Toxins is great, but again, the main lab has shutters over doors, including the maintenance door. None of these rooms should be a fort, and we need to design with fallibility in mind. If someone wants to risk breaking in or someone with access wants in, you shouldn't need thermite, teleportation, or creative usage of disposals to do it.

Swapping the xenobio sat and the testing sat was a great change. I approve.

**Tags and Other Nonsense**
I'm not testing your disposals loop because I hate disposals.  Hopefully it Just WorksTM.

The sec post is missing a camera and an air alarm.

The c_tag in the research room has a typo. (Reasearch) It's also missing an air alarm and a lightswitch.

The mech bay doesn't have a camera or air alarm. The light switch is diagonally placed when it could be cardinally placed. The warning stripe in the little maintenance area is facing the wrong way (The hallway is not dangerous)

The Science Hallway - Foyer camera may be redundant. There are five cameras in that hallway. You also don't need to have that mule delivery space so fortified.

The misc toxins area doesn't have an air alarm.

You should rename the camera "Science Gun Range" as "Science Firing Range" for consistency.

I was going to say the RD's room doesn't need two cameras, but bravo on the tinted windows. Very good design and a nice hiding spot if that camera is disabled.

image
Consider making these walls regular walls, or at least the ones into the foyer.

You can probably shorten the walkway to xenobio siginficantly by shifting solars over a few tiles and moving the lab up and left a few tiles.

I won't review the Xenobio sat because I assume nothing much was touched there.

Make sure you run the debug in case I missed something.

**Comment by Cartographer-D** [comment]
Just noticed something else as I was about to close out the application:
image
This wall right here is reinforced, but it could be a great bit of fun to have it be a regular wall instead. "Oh no, someone broke that wall and all my incredibly hot toxic gases are escaping into maintenance!"

**Comment by AlinaStarkova** [comment]
Current iteration, made nicer looking

image

**Comment by Cartographer-D** [comment]
Looks good.

I see a few minor edits to make:
1. The Research division server should be an autoname APC. "west"

2. There are some pixelshifts to do in xenobio, the nanite room, etc.:
image
Toxins alt room has a syringe taped to the side
image
Same with the research room
image
This pen is balancing on the precipice of its bin instead of laying inside
image
Hovering box in nanites
image
Hovering box in cytology
image
Fused box in xenobio

3. This is a weird change to the tinting of the windows. Do it in the middle, not at all, or fully I think.
image
